bot-samples/QR-SMS-Sheets-template/README.md

```diff
@@ -0,0 +1,76 @@
+
+As an example, we'll build this out for a venue which wants to attract repeat customers for an underground music night.
+## Prerequisites
+A basic familiarity with LivePerson Bots is assumed. However, relevant documentation is linked below for your convenience. As well, you can import the bot JSON for this project to save you the trouble of writing the dialogues and building the integration.
+
+- [Create and deploy a bot](https://developers.liveperson.com/tutorials-guides-getting-started-with-bot-building-overview.html) on SMS using the [default Twilio connector](https://knowledge.liveperson.com/getting-started-quick-start-guides-twilio-sms-quick-start.html)
+- [Create an SMS QR code](https://www.the-qrcode-generator.com/) pointing toward the Twilio number connected to the Conversational Cloud
+- Create a Google Sheet and a Google Developer account to [integrate into Conversation Builder](https://developers.liveperson.com/bot-accounts-credentials.html)
+- Ensure you have access to [Proactive Messaging](https://knowledge.liveperson.com/messaging-channels-proactive-messaging-proactive-messaging-user-guide.html)
+
+## Workflow
+Once we've got a [basic bot deployed to SMS](https://developers.liveperson.com/conversation-builder-testing-deployment-deploying-to-conversational-cloud.html), we can use the free tool linked above to generate the QR code, and we'll use [our Twilio SMS number](https://knowledge.liveperson.com/getting-started-quick-start-guides-twilio-sms-quick-start.html) which is connected to the Conversational Cloud as the target.
+
+![qr-code](./readme-images/qr-code.png?raw=true)
+
+At our imaginary music venue, we can put up posters with a QR code kicking off an SMS conversation:
+
+![poster](./readme-images/poster.png?raw=true)
+
+You can scan the QR code yourself to get a sense of the user experience, but essentially the QR code automatically:
+- Opens the consumer's SMS messaging app
+- Fills in the target phone number
+- Fills in the message text
+
+The consumer just has to hit send.
+
+![sms1](./readme-images/sms1.png?raw=true)
+
+Now, as each new consumer comes in, we want to store their phone numbers so that we can proactively reach out to them at a later date.
+
+First, in a Conversation Builder interaction, we'll grab the consumer's phone number and write it to a bot variable:
+
+```
+var customerInfo = botContext.getLPCustomerInfo();
+var phoneNumber = customerInfo.imei;
+botContext.setBotVariable('number', phoneNumber, true, false);
+```
+
+For long-term storage, we could use LivePerson's Conversation Context Service, an external database, a CRM, etc. In this case, we'll use a Google Sheet so we put have the data in exactly the right format to upload to the LivePerson Proactive Messaging tool.
+
+Our integration with Google Sheets will send a POST request to `https://sheets.googleapis.com/v4/spreadsheets/{$botContext.sheetId}/values/A1:append` with the consumer's phone number.
```

Comment:
I would put in some kind of link to how to create credentials, as they will be required the use the Google Sheets API. Additionally, they will need to create those credentials through a Google Developer Account. I had previously created a walkthrough of creating an account for this use case, so it might be helpful to include.
[Google Developers Signup (1).pdf](https://github.com/LivePersonInc/ConversationBuilder-Samples/files/7164578/Google.Developers.Signup.1.pdf)

Reply:
Can we brainstorm a way we can avoid making users integrate with another unrelated service such as sheets API here? mock APIs or a NoSQL solution should work here, creative use of JSON objects could be an option. Something to self contain the example, and able to just be passed off and used

Reply:
There's definitely no real need to use Sheets. I, personally, decided on it as then the data for Proactive Messaging would be in the same format as the upload requires. 

As well, Sheets has been handy in the past when doing demos as it's a "database" that laypeople understand, so I thought it might be handy to give folks some boilerplate to get started there.

However, if y'all think it muddies the water here I could remove Sheets and just use the Context Store. The thing is: how can the brand easily get the data from the Context Store and fire off a proactive messaging blast? I reckon the easiest way would be:
- Rig up a script (I'd use a Node.js script as it's what I'm most familiar with) 
- Grab the data from the Context Store APIs
- Put the data into the right format for the Proactive Messaging API
- Use the Proactive Messaging API to send the blast
It would be a bit of work to re-architecture the solution this way, but I think it could be useful (maybe more useful than the Sheets example).

I reckon a comprehensive bot template that hooks into Sheets would be handy, but it could be de-coupled from QR codes, SMS, proactive messaging, etc.

Let me know your thoughts @dahby @JasonSowers @jaltschuler  and I'll proceed accordingly.

Reply:
@cshape I think that the script/API approach would be appropriate. I'd recommend that we do not spend too much of the guide explaining the script, but maybe linking to a different resource in this repository that holds the code for the script to ensure that it doesn't muddy the waters of the guide too much. Say something like, "Here's how I constructed the script, but you could do something similar with your own database solution (Google sheets, NoSQL, etc)". I think as long as the focus remains on the QR/data storage/proactive flow, that this guide serves its purpose.

Reply:
Sounds good. I've almost finished the script and will make another commit today with the recommended changes.

@dahby @JasonSowers @jaltschuler

Reply:
Made a new commit with the recommended changes. Please check 'er out @dahby @JasonSowers

The script is a bit of a beast but it does work and I think I made it simple to understand. :)